File: sample/src/main/java/com/schibsted/spain/barista/sample/HintAndErrorActivity.java
package com.schibsted.spain.barista.sample;

import android.os.Bundle;
import androidx.annotation.Nullable;
import com.google.android.material.textfield.TextInputEditText;
import com.google.android.material.textfield.TextInputLayout;
import androidx.appcompat.app.AppCompatActivity;
import android.widget.EditText;

public class HintAndErrorActivity extends AppCompatActivity {
  @Override
  protected void onCreate(@Nullable Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_hintanderrortext);

    showError();
  }

  private void showError() {
    TextInputLayout inputLayout = findViewById(R.id.hintanderror_inputlayout);
    inputLayout.setError(getString(R.string.hintanderror_inputlayout_error));

    TextInputEditText inputEditText = findViewById(R.id.hintanderror_inputedittext);
    inputEditText.setError(getString(R.string.hintanderror_inputedittext_error));

    EditText editText = findViewById(R.id.hintanderror_edittext);
    editText.setError(getString(R.string.hintanderror_edittext_error));
  }
}

File: sample/src/main/java/com/schibsted/spain/barista/sample/HintAndErrorActivity.kt
package com.schibsted.spain.barista.sample

import android.os.Bundle
import android.view.View
import android.widget.EditText
import androidx.appcompat.app.AppCompatActivity
import com.google.android.material.textfield.TextInputEditText
import com.google.android.material.textfield.TextInputLayout

class HintAndErrorActivity : AppCompatActivity() {
  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_hintanderrortext)
    findViewById<View>(R.id.showErrors).setOnClickListener { v: View? -> showError() }
  }

  private fun showError() {
    val inputLayout = findViewById<TextInputLayout>(R.id.hintanderror_inputlayout)
    inputLayout.error = getString(R.string.hintanderror_inputlayout_error)
    val inputEditText = findViewById<TextInputEditText>(R.id.hintanderror_inputedittext)
    inputEditText.error = getString(R.string.hintanderror_inputedittext_error)
    val editText = findViewById<EditText>(R.id.hintanderror_edittext)
    editText.error = getString(R.string.hintanderror_edittext_error)
  }
}